clang/lib/CodeGen/CGAtomic.cpp

```diff
@@ -1453,9 +1457,11 @@ void AtomicInfo::EmitAtomicLoadLibcall(llvm::Value *AddForLoaded,
 }
 
 llvm::Value *AtomicInfo::EmitAtomicLoadOp(llvm::AtomicOrdering AO,
-                                          bool IsVolatile) {
+                                          bool IsVolatile, bool CastFP) {
   // Okay, we're doing this natively.
-  Address Addr = getAtomicAddressAsAtomicIntPointer();
+  Address Addr = getAtomicAddress();
+  if (!(Addr.getElementType()->isIEEELikeFPTy() && !CastFP))
```

Comment:
```suggestion
  if (CastFP || !Addr.getElementType()->isIEEELikeFPTy())
```

Reply:
Still should probably pass through the non-ieee type case through to the IR also, but step at a time I guess